internal/olm/client/client.go

```diff
@@ -248,25 +264,39 @@ func (c Client) printDeploymentErrors(ctx context.Context, key types.NamespacedN
 		}
 		depSelectors := ds.Spec.Selector
 		if err := c.KubeClient.Get(ctx, depKey, dep); err != nil {
-			log.Printf("error getting operator deployment %q: %v", ds.Name, err)
+			depErr = append(depErr, deploymentError{
+				depName: ds.Name,
+				issue:   err.Error(),
+			})
 			continue
 		}
 		for _, s := range dep.Status.Conditions {
 			if s.Type == appsv1.DeploymentAvailable && s.Status == corev1.ConditionFalse {
```

Comment:
Why we check if `appsv1.DeploymentAvailable` ? Should not return an error if we have any failure conditional?

Reply:
`Available=False` always indicates that deployment has failed, This condition is checking to ensure this.

Reply:
But it is confuse and shows not accurate see:

// Get Deployment
// For all conditions if the deployment 
//.       - Check if is Avaliable and Status is == conditionFalse

Should not be: 

// Get Deployment
// If Avaliable 
// Get all conditionals
// Check if any contional is == a failure

Reply:
I cacher you are check if the specific conditional DeploymentAvaible is == ConditionFalse. 
IMO it should be != of success since unknown means failure as well.